File: js/examples/manageHostedDnsRecords.js
import { StackMachine } from "stackmachine";

const STACKMACHINE_API_KEY = process.env.STACKMACHINE_API_KEY;

if (!STACKMACHINE_API_KEY) {
  throw new Error("Set STACKMACHINE_API_KEY before running this example.");
}

const client = new StackMachine(STACKMACHINE_API_KEY);

const dnsDomainId = "dns_domain_example";

const records = await client.dns.records.list({ domain: dnsDomainId });
console.log("Hosted DNS records:", records);

const record = await client.dns.records.create({
  domain: dnsDomainId,
  kind: "A",
  name: "www",
  value: "192.0.2.1",
  ttl: 300,
});
console.log("Created DNS record:", record.id);

const updated = await client.dns.records.update(record.id, {
  domain: dnsDomainId,
  kind: "A",
  name: "www",
  value: "192.0.2.2",
  ttl: 300,
});
console.log("Updated DNS record:", updated.id);

// Uncomment to delete the DNS record created by this example.
// await client.dns.records.del(record.id);

File: python/examples/manage_hosted_dns_records.py
import os

from stackmachine import StackMachine

dns_domain_id = "dns_domain_example"

with StackMachine(os.environ["STACKMACHINE_API_KEY"]) as client:
    records = client.dns.records.list(domain=dns_domain_id)
    print("Hosted DNS records:", records)

    record = client.dns.records.create(
        domain=dns_domain_id,
        kind="A",
        name="www",
        value="192.0.2.1",
        ttl=300,
    )
    print("Created DNS record:", record.id)

    updated = client.dns.records.update(
        record.id,
        domain=dns_domain_id,
        kind="A",
        name="www",
        value="192.0.2.2",
        ttl=300,
    )
    print("Updated DNS record:", updated.id)

    # Uncomment to delete the DNS record created by this example.
    # client.dns.records.delete(record.id)
